analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyToAutoProperty.cs

```diff
@@ -158,8 +148,8 @@ private static bool TryGetFieldFromSetter(AccessorDeclarationSyntax setter, Sema
                 return false;
             }
 
-            if (!(semanticModel.GetSymbolInfo(identifier).Symbol is INamedTypeSymbol type) ||
-                !type.Equals(declaringType))
+            if (!(semanticModel.GetSymbolInfo(identifier).Symbol is INamedTypeSymbol type)
```

Comment:
This whole method is written just wrong and with a wrong name, forcing the wrong design in the rest of the rule. Let's change the signature to 
```
private static bool FieldSymbol(ExpressionSyntax expression, INamedTypeSymbol declaringType, SemanticModel semanticModel)
```